src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +249,62 @@ public static Path of(URI uri) {
      */
     Path getFileName();
 
+    /**
+     * Returns the file name extension of this path as a {@code String}. The
+     * extension is the portion of the {@linkplain #getFileName file name}
+     * string after the last period character ('.', U+002E FULL STOP). Leading
+     * period characters in the string are ignored. If there is no  extension,
```

Comment:
The definition of the extension is in two sentences, first this:

> The extension is the portion of the file name string after the last period character.

Followed by this:

> Leading period characters in the string are ignored.

This is confusing, since the first sentence reads like a complete definition. But the second sentence modifies it, forcing the reader to go back and reinterpret.

I'd suggest putting the definition into a single sentence, something like this:

> The extension is the portion of the _file name_ string
> after the last period character (...) that is not a leading
> period character in the string.

(I've omitted the link and the Unicode callout here for ease of reading, but I think they should be
included in the actual markup, as before.)

(Also, to my ear "dot" reads better than "period" since conventionally a filename like `abc.txt` would be read aloud as "aye bee see dot tee ex tee". But it's ok if you want to stick with period consistently.)